### ISSUE TITLE
Group services together based on host

### ISSUE THREAD

**Issue by steaksauce-** [Icinga/icingaweb2#4099]
## Is your feature request related to a problem? Please describe.
It would be useful to group unhandled services according to their host, making things easier to read.

## Describe the solution you'd like
Group unhandled services (or services in general) according to their host. Examples of this can be seen in the Icinga 1.x screenshot below, and Thruk screenshot.

## Describe alternatives you've considered
I do not have an alternative to this. The only deviation I can think of it having it as a toggle-able feature

## Additional context
**Screenshot of service grouping in Icinga 1.x**
![Screen Shot 2020-03-05 at 4 09 21 PM](https://user-images.githubusercontent.com/8814619/76031640-b125c200-5efd-11ea-89c2-973787e2450d.png)

**Screenshot of service grouping in Thruk**
![Screen Shot 2020-03-05 at 4 10 03 PM](https://user-images.githubusercontent.com/8814619/76031625-abc87780-5efd-11ea-9397-0be5ce53938b.png)

**Screenshot of NO service grouping in Icingaweb2**
_Believe me, the services circled in blue belong to the same host :)_
![Screen Shot 2020-03-05 at 4 10 39 PM](https://user-images.githubusercontent.com/8814619/76031592-9a7f6b00-5efd-11ea-9162-c19105899225.png)

**Comment by nilmerg** [comment]
Hi,

implementing this visually in those lists is a rather enormous change. That's why we will probably not implement this for the monitoring module. (Such adjustments are more likely in Icinga DB Web)

Especially since there is a different view which perfectly covers the possibility to quickly see hosts and services in their relationship and to easily identify issues relating to a single host. The servicegrid:

![Bildschirmfoto von 2020-05-13 14-05-31](https://user-images.githubusercontent.com/16668527/81810210-def13d80-9522-11ea-8dc5-f588d1cfe840.png)

For [Icinga DB Web](https://github.com/Icinga/icingadb-web) however, we might think of a solution to implement this. Please open a new issue there if the servicegrid is not already the solution for you. Thanks.

**Comment by steaksauce-** [comment]
The service grid doesn't supply the same level of detail or control as the "service problems" view, and is clunky when you have 1000s of unique services. ie, a user cannot select all of the services in a host and manipulate both (say adding scheduled downtime), in addition to service output not being readily available.

Icinga DB Web isn't a solution as it would be yet another UI we would have to introduce and potentially switch back and forth between.

I would think that visually displaying them grouped together would be relatively simple since services MUST belong to a host, and you can already view the services independently in a list by clicking on the host.

**Comment by nilmerg** [comment]
The servicegrid provides way more information regarding the possibility to *pick* a problematic host or a range of problematic services. If you have plenty of services, use the filter to narrow them down to what you'll want to see. By clicking on a hostname in the grid you can very well see all services belonging to that host and perform actions on them. Service output is also there, by hovering the service bubble in question.

So please give the servicegrid a real chance. It's fine that you've got your workflow in Icinga Classic UI and Thruk, but to be honest with you, these interfaces are a decade old and didn't survive Icinga 2 for a reason. It's not that Icinga Web 2 doesn't provide you the same control and more after all, it's just presented differently.

**Comment by steaksauce-** [comment]
TLDR: because of our (probably unique) situation, service grid is neat, but does not fit any of our needs. Even a line spanning the screen to separate hosts on the service problems list when we sort by hostname would work for us. Keep in mind that I'm just the monitoring "engineer/admin", not the actual end user, which is why I refer to this problem as a "we/us" problem.

-------------------------------------------------
The problem with the service grid is we have 5.5k+ hosts and ~20k services, a good amount of the services are unique to a given host.

There is a toggle button to only show problems, but for most hosts I have to scroll 20 or 30 pages before I even see the problem.

A grid is not good for this scenario, a list is.

In addition to this, the grid does not give us detailed status output without clicking the box. In addition, users cannot select multiple services from the grid to apply the same acknowledgment to them (our acknowledgment comments are based off of a ticket #). Even if we could select multiple, you're still scrolling 20 or 30 pages left-right or up-down to find everything you need.

We aren't asking for the old UI back. We are simply asking for the grouping of services under a given host in a view -- in this case we only care about the service problems view.

The root problem is that during multiple outages, some of the services under a host get lost because the default sorting appears to be time (and depending on the service, the check interval could be 1, 3, or 5 minutes in our case). We can get pretty close to what we need by sorting by hostname instead, but because there's no visible separator between the hosts, sometimes it all runs together and introduces a higher rate of human error.

**Comment by nilmerg** [comment]
Okay, fine, you say your environment is a very heterogeneous one. Indeed, in such a case the grid will spread the service columns/rows too far for it being helpful.

The thing with a simple separator in a list is, it's an *alien*. It stands out too much very quickly and we have to decide when to show it, probably even so that the user (you) can control/influence it somehow. Let's use the example you've given, sorting by `hostname`. In the current monitoring module, this would only be helpful if there's a possibility to also sort by another *second* column. e.g. `hostname` and `service_severity` which sorts by `hostname` first and then these subsets again by `service_severity`. The user would need to be able to choose in both cases by what to order and which one is first or second. The key here is to allow enough flexibility for the user when implementing this. That's why I mentioned Icinga DB Web, because there this differentiation is already possible, though only technically and not in the UI (yet).

Just displaying a separator if the list is sorted by `hostname` is a short-sighted solution. Implementing a proper solution for the monitoring module won't happen though, because we focus on the successor (Icinga DB Web) for such big features instead. That's why I want to avoid it all cost and try to find another solution.

So I still think there is an alternative. You wrote:

> We aren't asking for the old UI back. We are simply asking for the grouping of services under a given host in a view -- in this case we only care about the service problems view.

![Bildschirmfoto von 2020-05-18 09-08-15](https://user-images.githubusercontent.com/16668527/82183972-2b53c900-98e7-11ea-9bb9-09ed675a1598.png)

That's the `Services` tab in a host or service detail view. So your users see a problem in the list (be it the host or service list, doesn't matter) and then click on the problem in question. If it's necessary to perform actions on all services then, the `Services` tab provides all the features.
